hal/modem_api.h

```diff
@@ -0,0 +1,88 @@
+/* mbed Microcontroller Library
```

Comment:
I like this: it allows modem power-on/off (and potentially other modem functionality) to be abstracted neatly. For instance, we have many 10s of modems that each require different power on pulsing sequences and this way, we get to control that just like we would any other HW block.